src/xdp/xsk.c

```diff
@@ -3905,6 +4128,16 @@ XskNotifyValidateParams(
         return STATUS_INVALID_DEVICE_STATE;
     }
 
+    if (RequestorMode != KernelMode && *CompletionContext != NULL) {
+        return STATUS_INVALID_PARAMETER;
+    }
+
+    if (UseCallback && *TimeoutMilliseconds > 0) {
```

Comment:
i might just assert this for kernel mode callers, since this routine is a hot code path